requests/auth.py

```diff
@@ -27,7 +27,21 @@
 
 def _basic_auth_str(username, password):
     """Returns a Basic Auth string."""
-    
+
+    # "I want us to put a big-ol' comment on top of it that
+    # says that this behaviour is dumb but we need to preserve
+    # it because people are relying on it."
+    #    - Lukasa
+    #
+    # These are here solely to maintain backwards compatibility
+    # for things like ints. This will be removed in 3.0.0.
+    if not isinstance(username, basestring):
```

Comment:
Honestly, a whole lot of stuff got removed that doesn't have deprecation warnings. I kinda think that ship has sailed.